talk/owt/sdk/base/sdputils.cc

```diff
@@ -192,29 +194,70 @@ std::string SdpUtils::SetPreferCodecs(const std::string& sdp,
+    before_strip =
+        std::regex_replace(sdp, reg_m_line, m_line_stream.str());
+    after_strip = before_strip;
+
+    if (!is_audio && set_quality) {
+      // Search for c-line and add a=priority line after it.
+      std::regex reg_c_line("c=IN .*\\r\\n");
+      std::smatch c_line_match;
+      auto cline_search_result =
+          std::regex_search(after_strip, c_line_match, reg_c_line);
+      if (!cline_search_result || c_line_match.size() == 0) {
+        RTC_LOG(LS_WARNING) << "c-line is not found. SDP: " << sdp;
+        return sdp;
+      }
+      std::string c_line(c_line_match[0]);
+      std::stringstream c_line_stream(c_line);
+      std::string new_cline = c_line_stream.str() + "a=quality:10\r\n";
+
+      after_strip = std::regex_replace(before_strip, reg_c_line, new_cline);
       before_strip = after_strip;
-      std::regex reg_rtcp_map(
-          "a=rtcp-fb:" + codec_value + " .*\\r\\n",
-          std::regex_constants::icase);
-      after_strip = std::regex_replace(before_strip, reg_rtcp_map, "");
+    }
+    // Remove all a=fmtp:xx, a=rtpmap:xx and a=rtcp-fb:xx where xx is not in
+    // m-line, this includes the a=fmtp:xx apt:yy lines for rtx.
+    for (size_t i = 3, m_line_vector_size = m_line_vector.size();
+         i < m_line_vector_size; i++) {
+      if (std::find(kept_codec_values.begin(), kept_codec_values.end(),
+                    m_line_vector[i]) == kept_codec_values.end()) {
+        std::string codec_value = m_line_vector[i];
+        std::regex reg_rtp_xx_map("a=rtpmap:" + codec_value + " .*\\r\\n",
+                                  std::regex_constants::icase);
+        after_strip = std::regex_replace(before_strip, reg_rtp_xx_map, "");
+        before_strip = after_strip;
+        std::regex reg_fmtp_xx_map("a=fmtp:" + codec_value + " .*\\r\\n",
+                                   std::regex_constants::icase);
+        after_strip = std::regex_replace(before_strip, reg_fmtp_xx_map, "");
+        before_strip = after_strip;
+        std::regex reg_rtcp_map("a=rtcp-fb:" + codec_value + " .*\\r\\n",
+                                std::regex_constants::icase);
+        after_strip = std::regex_replace(before_strip, reg_rtcp_map, "");
+        before_strip = after_strip;
+      }
+    }
+  } else {
+    if (!is_audio && set_quality) {
+      // Search for c-line and add a=priority line after it.
+      std::regex reg_c_line("c=IN .*\\r\\n");
+      std::smatch c_line_match;
+      auto cline_search_result =
+          std::regex_search(after_strip, c_line_match, reg_c_line);
+      if (!cline_search_result || c_line_match.size() == 0) {
+        RTC_LOG(LS_WARNING) << "c-line is not found. SDP: " << sdp;
+        return sdp;
+      }
+      std::string c_line(c_line_match[0]);
+      std::stringstream c_line_stream(c_line);
+      std::string new_cline = c_line_stream.str() + "a=quality:10\r\n";
+
+      after_strip = std::regex_replace(before_strip, reg_c_line, new_cline);
```

Comment:
It's easy to write unit tests for `SdpUtils`, please consider to add unit tests.

Reply:
We will create a seperate "huge" pr for enabling test cases as follow up.

Reply:
FYI, test framework is ready, entry is [here](https://github.com/open-webrtc-toolkit/owt-client-native/blob/master/talk/owt/sdk/test/unittest_main.cc). Don't forget to enable unit tests in CI. I guess unit tests are disabled by QA.